src/modules/graph/graph.service.ts

```diff
@@ -59,13 +68,45 @@ export class GraphService {
     try {
       return await request.post(requestBody);
     } catch (error) {
-      createWrapGraphError({
+      createGraphError({
         error,
         messageForUnknownError: 'An unexpected error occurred.',
         knownErrors: [postFacilityTermExistsKnownError()],
       });
     }
   }
+
+  async patch<RequestBody, ResponseBody>({ path, requestBody }: GraphPatchParams<RequestBody>): Promise<ResponseBody> {
+    const request = this.createPatchRequest({ path });
+    return await this.makePatchRequest({ request, requestBody });
+  }
+
+  private createPatchRequest({ path }: { path: string }): GraphRequest {
+    return this.client.api(path);
+  }
+
+  private async makePatchRequest<RequestBody>({ request, requestBody }: { request: GraphRequest; requestBody: RequestBody }) {
+    try {
+      return await request.patch(requestBody);
+    } catch (error) {
+      createGraphError({
+        error,
+        messageForUnknownError: 'An unexpected error occurred.',
+        knownErrors: [],
+      });
+    }
+  }
+
+  async uploadFile(file: NodeJS.ReadableStream, fileSizeInBytes: number, fileName: string, urlToCreateUploadSession: string): Promise<UploadResult> {
+    const uploadSession: LargeFileUploadSession = await LargeFileUploadTask.createUploadSession(this.client, urlToCreateUploadSession, {});
```

Comment:
@avaitonis The object that we pass as the last argument allows us to set options, notably the behaviour when a file with the same name already exists in the location. What do we want this to be? The options are overwrite the existing file, rename the new file `conflictingFileName N.extension` when it is the Nth file with that name, or fail (which returns a `409` error - would we want to map this to a `400`?). 

In the Mulesoft code, it appears that in the SharePoint connector `file-add` operation the `Overwrite` boolean isn't set, and the default is false, but the docs don't make it clear what happens in this case (i.e., does it fail or rename the new file?) https://docs.mulesoft.com/sharepoint-connector/3.5/sharepoint-connector-reference#fileAdd

Reply:
Agree that in Mulesoft it is not overriding file, I don't remember this being an issue.

I could vote for renaming file with N+1th. Fail is also fine as it should highlight issues.

@abhi-markan what do you think should happen if file exists in Sharepoint?

Reply:
We shouldn't override the file if it already exists. this is because Underwriters could make changes to a file with a similar name in SP and we could override their work. 

Let's use the 400 error - this is because I wouldn't want to have a lot of files with similar names to exist

Reply:
Thanks @IgnatG, I've made this change